docs/specs/auth-plugin.md

```diff
@@ -0,0 +1,73 @@
+<script src="https://gist.github.com/rgwozdz/4098e0e18cc85299f4444a3e64a7c68b.js"></script>
+
+
+### Function: getAuthenticationSpecification
+
+Authorization plugins are required to return a function called "getAuthenticationSpecification".  Its purpose is to configure a function that is used to describe the authentication specification. It should have the following signature:
+
+ ##### getAuthenticationSpecification(providerNamespace) ⇒ function
+ 
+ Param | Type | Description |
+| --- | --- | --- |
+| providerNamespace | <code>string</code> | a provider's namespace |
+
+As noted, `getAuthenticationSpecification` returns a function configured with a provider's namespace. The configured function should return an object with the following specification: 
+
+<script src="https://gist.github.com/rgwozdz/194106328acd4d32fbbdb2b88c1c866d.js"></script>
+
+An example of `getAuthenticationSpecification` is available [here](https://github.com/koopjs/koop-auth-direct-file/blob/master/src/index.js#L44-L56).
+
+
+### Function: authenticate
+
+Authorization plugins are required to return a function called `authenticate`.  Its purpose is to validate credentials and, if successful, issue a token for authorizing subsequent resource requests.  The `authenticate` function should have the following signature:
+
+##### authenticate(username, password) ⇒ Promise
+
+| Param | Type | Description |
+| --- | --- | --- |
+| username | <code>string</code> | requester's username |
+| password | <code>string</code> | requester's password |
+
+
+As noted above, the `authenticate` function should return a promise. If the authentication is unsuccessful the promise should reject with an error object.  The error should have a `code` property with value `401`. If the authentication is successful, the promise should resolve an object with the following properties:
+
+<script src="https://gist.github.com/rgwozdz/283d8604676acd972fb72aef9a228033.js"></script>  
+
+Authorization plugins are free to validate credentials in any manner.  For example, you might check a database for a match of the submitted username and password, or forward the credentials on to a third-party identity-store. [koop-auth-direct-file](https://github.com/koopjs/koop-auth-direct-file) provides an [example](https://github.com/koopjs/koop-auth-direct-file/blob/master/src/index.js#L59-L88) of a very basic credential validation using a simple JSON file-store. 
+
+### Function: authorize
+
+Authorization plugins are required to return a function called `authorize`.  Its purpose is to validate a token submitted with a resource request.  It should have the following function signature:
+
+##### authorize(token) ⇒ Promise
+
+| Param | Type | Description |
+| --- | --- | --- |
+| token | <code>string</code> | token that can be used to prove previously successful authentication |
```

Comment:
does this have to be a token or can it be any object that the auth plugin understands?

Reply:
Nice catch. It doesn't have to be a token.